### PR TITLE
fix docs.rs show features

### PR DESCRIPTION
**Pull request by light4**
use [crates.io](https://crates.io) to get deps info

I don't remember how it looks like before, any advice?

### CODE REVIEW

**Comment by Folyd** [comment]
Thanks, @light4. I ever didn't aware the feature flags were broken. 😅

File: manifest.jsonnet [hunk 1]
```diff
@@ -46,7 +46,7 @@ else
 
 local INDEX_MANAGER_FILES = ['core/storage.js', 'index-manager.js'];
 json.addIcons(icons())
-.addPermissions(['storage', 'unlimitedStorage'])
+.addPermissions(['storage', 'unlimitedStorage', 'webRequest', 'webRequestBlocking', '*://crates.io/api/v1/crates/*'])
```
Comment (Folyd) [inline manifest.jsonnet:49]: We use Manifest V3 already, after loading into Chrome, it said: `webRequestBlocking' requires manifest version of 2 or lower`. Actually, I think we can remove those permissions.

File: extension/script/docs-rs.js [hunk 2]
```diff
@@ -116,8 +116,9 @@ document.addEventListener("DOMContentLoaded", async () => {
 async function enhanceFeatureFlagsMenu(menu) {
     // Use rawCrateName to fetch the Cargo.toml, otherwise will get 404.
     let cargoTomUrl = `https://docs.rs/crate/${rawCrateName}/${crateVersion}/source/Cargo.toml`;
-    let response = await fetch(cargoTomUrl);
-    let content = await response.text();
+    let crateAPIURL = `https://crates.io/api/v1/crates/${rawCrateName}/${crateVersion}`;
```
Comment (Folyd) [inline extension/script/docs-rs.js:119]: The `crateVersion` could be `latest` which is invalid for the API of crates.io. For example: https://crates.io/api/v1/crates/tokio/latest

We should ensure the `enhanceFeatureFlagsMenu()` function is called after `crateVersion = parseCrateVersionFromDOM()`, see line 97.

**Review by Folyd — CHANGES_REQUESTED** [review]
Overall, it looks good to me. Some improvement is needed, though. 

In case you don't remember the old UI,  here is the screenshot. It's up to you to keep the same or improve it.
image

**Comment by Folyd** [comment]
Also, don't forget to remove `tests/script-lib.sepc.js`.

**Comment by light4** [comment]
Updated, should I rebase to one commit?

File: manifest.jsonnet [hunk 3]
```diff
@@ -46,7 +46,7 @@ else
 
 local INDEX_MANAGER_FILES = ['core/storage.js', 'index-manager.js'];
 json.addIcons(icons())
-.addPermissions(['storage', 'unlimitedStorage'])
+.addPermissions(['storage', 'unlimitedStorage', '*://crates.io/api/v1/crates/*'])
```
Comment (Folyd) [inline manifest.jsonnet:49]: We can remove `*://crates.io/api/v1/crates/*` too.
> Permission '*://crates.io/api/v1/crates/*' is unknown or URL pattern is malformed.

File: extension/script/docs-rs.js [hunk 4]
```diff
@@ -114,10 +114,14 @@ document.addEventListener("DOMContentLoaded", async () => {
 });
 
 async function enhanceFeatureFlagsMenu(menu) {
+    if (crateVersion === 'latest') {
+        crateVersion = parseCrateVersionFromDOM();
+    }
     // Use rawCrateName to fetch the Cargo.toml, otherwise will get 404.
     let cargoTomUrl = `https://docs.rs/crate/${rawCrateName}/${crateVersion}/source/Cargo.toml`;
```
Comment (Folyd) [inline extension/script/docs-rs.js:121]: The `cargoTomUrl` is useless now.

**Review by Folyd — CHANGES_REQUESTED** [review]
Just some nitpicking.

**Comment by Folyd** [comment]
> Updated, should I rebase to one commit?

I'll squash it into one commit.

File: manifest.jsonnet (hunk 3 above)
Comment (light4) [reply manifest.jsonnet:49]: Without this, Firefox says `TypeError: NetworkError when attempting to fetch resource.`

Comment (light4) [reply manifest.jsonnet:49]: I found this, will make an update.
https://www.extension.ninja/blog/post/solved-permission-is-unknown-or-url-pattern-is-malformed/

Comment (Folyd) [reply manifest.jsonnet:49]: Good to know. Thanks.

File: extension/script/docs-rs.js (hunk 4 above)
Comment (Folyd) [reply extension/script/docs-rs.js:121]: Ok, thanks for reminding me.

**Comment by Folyd** [comment]
I'll merge it. We can fix the permission issue in the next PR.

**Pull request merged**